R/dataframe-frame.R

```diff
@@ -1888,3 +1888,49 @@ dataframe__write_parquet <- function(
     invisible(self)
   })
 }
+
+#' Sample from this DataFrame
+#'
+#' @inheritParams expr__sample
+#' @inherit as_polars_df return
+#' @examples
+#' df <- pl$DataFrame(
+#'   foo = 1:3,
+#'   bar = 6:8,
+#'   ham = c("a", "b", "c")
+#' )
+#' df$sample(n = 2, seed = 0)
+dataframe__sample <- function(
+  n = NULL,
+  ...,
+  fraction = NULL,
+  with_replacement = FALSE,
+  shuffle = FALSE,
+  seed = NULL
+) {
+  wrap({
+    check_dots_empty0(...)
+    if (!is.null(fraction) && !is.null(n)) {
+      abort("cannot specify both `n` and `fraction`")
+    }
+    if (is.null(seed)) {
+      seed <- sample.int(10000, 1)
+    }
```

Comment:
Unclear to me why the Rust part has `seed: Option<u64>` since we cannot missing `seed` but it is what it is.

Reply:
My guess is that it is because we want to control the value on the Rust side by the Seed setting on the bindings side?
I think we can confirm this behavior with a unit test.

Reply:
> we want to control the value on the Rust side by the Seed setting on the bindings side?

I don't understand what this means. My point was just that if on the binding side we replace NULL/None by a value, then `seed` should just be `u64` and not `Option<u64>` in Rust. But that's not very important, I've added a test that just checks there's no error when we let `seed = NULL`